File: Backend/app/src/main/java/com/example/demo/model/Video.java
package com.example.demo.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

@Entity
@Table(name = "videos")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Video {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;
    private Long channelId;
    private Long PlaylistId;

    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String url;
    private String urlThumbnail;

    private String status;        // draft / processing / published / private / deleted / blocked
    private String visibility;    // public / unlisted / private

    private Long publishedAt;     // timestamp
    private Long createdAt;       // timestamp
    private Long updatedAt;       // timestamp

    @PrePersist
    public void onCreate() {
        this.createdAt = Instant.now().toEpochMilli();
        this.updatedAt = Instant.now().toEpochMilli();
        if (this.status == null) this.status = "draft";
        if (this.visibility == null) this.visibility = "public";
    }

    @PreUpdate
    public void onUpdate() {
        this.updatedAt = Instant.now().toEpochMilli();
    }
}

File: Mobile/app/src/main/java/com/example/myapplication/data/model/Video.kt
package com.example.myapplication.data.model

import kotlinx.serialization.Serializable

@Serializable
data class Video(
        val id: Long? = null,
        val userId : Long,
        val channelId : Long?=null,
        val playlistId : Long?=null,
        val title: String? = null,
        val description: String? = null,
        val url: String? = null,
        val urlThumbnail: String,
        val status: String? = null,        // draft / processing / published / private / deleted / blocked
        val visibility: String? = null,    // public / unlisted / private
        val publishedAt: Long? = null,     // timestamp
        val createdAt: Long? = null,       // timestamp
        val updatedAt: Long? = null        // timestamp
)